docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,104 @@
+# Charm Architecture
+
+At its core, Content-cache-k8s is a NGINX cache application that helps another charm serving static content for it.
+
+The charm design leverages the [sidecar](https://kubernetes.io/blog/2015/06/the-distributed-system-toolkit-patterns/#example-1-sidecar-containers) pattern to allow multiple containers in each pod with [Pebble](https://juju.is/docs/sdk/pebble) running as the workload container’s entrypoint.
+
+Pebble is a lightweight, API-driven process supervisor that is responsible for configuring processes to run in a container and controlling those processes throughout the workload lifecycle.
+
+Pebble `services` are configured through [layers](https://github.com/canonical/pebble#layer-specification), and the following containers represent each one a layer forming the effective Pebble configuration, or `plan`:
+
+1. An [NGINX](https://www.nginx.com/) container with caching features, which can be used to efficiently serve static resources, as well as be the incoming point for all web traffic to the pod. This is the content cache container itself.
+2. A [NGINX Prometheus Exporter](https://github.com/nginxinc/nginx-prometheus-exporter) container that can be used to provide statistics on web traffic.
+
+As a result, if you run a `kubectl get pods` on a namespace named for the Juju model you've deployed the Content-cache-k8s charm into, you'll see something like the following:
+
+```bash
+NAME                             READY   STATUS    RESTARTS   AGE
+content-cache-k8s-0              3/3     Running   0          97s
+
+```
+
+This shows there are 3 containers - the two named above, as well as a container for the charm code itself.
+
+And if you run `kubectl describe pod content-cache-k8s-0`, all the containers will have as Command ```/charm/bin/pebble```. That's because Pebble is responsible for the processes startup as explained above.
+
+## Containers
+
+Configuration files for the containers can be found in [the files directory of the charm repository](https://github.com/canonical/content-cache-k8s-operator/tree/main/files) and in [the templates directory of the charm repository](https://github.com/canonical/content-cache-k8s-operator/tree/main/templates)
+
+### Content Cache
+
+This container is the entry point for all web traffic to the pod (on port `80`). Serves static files directly.
```

Comment:
Only 80 or 443 as well?

Reply:
For now, the container port is only set to 80.